noccur.el

```diff
@@ -54,13 +54,26 @@ ls-files' and 'grep'."
   (interactive (occur-read-primary-args))
   (let* ((default-directory (read-directory-name "Search in directory: "))
          (files (mapcar #'find-file-noselect
-                        (noccur--find-files regexp))))
+                        (noccur--find-files-project regexp))))
+    (multi-occur files regexp nlines)))
+
+;;;###autoload
+(defun noccur-directory (regexp &optional nlines)
```

Comment:
thdox writes:

>> @@ -54,13 +54,26 @@ ls-files' and 'grep'."
>>    (interactive (occur-read-primary-args))
>>    (let* ((default-directory (read-directory-name "Search in directory: "))
>>           (files (mapcar #'find-file-noselect
>> -                        (noccur--find-files regexp))))
>> +                        (noccur--find-files-project regexp))))
>> +    (multi-occur files regexp nlines)))
>> +
>> +;;;###autoload
>> +(defun noccur-directory (regexp &optional nlines)

What is your use-case for non-recursive multi-occur?

> I've googled what are the elisp's most common way of naming this kind
> of things. What I've found is that when this is recursive, that
> keyword "recursive" is part of the function name.

Not necessarily.  In fact, most functions do not contain "recursive" in
their name.

> So I would propose as proposition 1:
> - `noccur-directory` --> `noccur-directory`
> - `noccur-project` --> `noccur-directory-recursive`
>
> As per the anecdote, I've found `noccur`, because I was looking for a `multi-occur` for files in a directory. I found strange, and did not understood at first why `noccur-project` was named like that, with "project" in its name. I understood later that this was because of projectile. 
>
> I could also propose as proposition 2:
> - `noccur-project` (recursive as much as the scope of the git project, falling back to `find` if no git/bzr/hg)
> - `noccur-directory-recursive` (not using `git ls-files`)
> - `noccur-directory` (not using `git ls-files`, not recursive)

What about `noccur-directory-non-recursive` instead?  The reason for it
is that I think it should be recursive by default.

Nico
-- 
Nicolas Petton
http://nicolas-petton.fr